source/constructs/api/data_source/glue_database_detector.py

```diff
@@ -20,95 +20,33 @@
         RoleArn=f"{iam_role_name}",
-        RoleSessionName="rds-instance-source-detection"
+        RoleSessionName="glue-database-source-detection"
     )
     credentials = assumed_role_object['Credentials']
     regions = crud.get_account_agent_regions(aws_account_id)
+    glue_database_list = []
     for region in regions:
-        total_rds = 0
-        connected_rds = 0
         client = boto3.client(
-            'rds',
+            'glue',
             aws_access_key_id=credentials['AccessKeyId'],
             aws_secret_access_key=credentials['SecretAccessKey'],
             aws_session_token=credentials['SessionToken'],
             region_name=region
         )
-        rds_agent_list = []
-        """ :type : pyboto3.rds """
-        logger.info("detect_rds_data_source")
-        for instance in client.describe_db_instances()['DBInstances']:
-            logger.info(instance)
-            if instance['Engine'] not in const.RDS_SUPPORTED_ENGINES:
-                continue
-            rds_agent_list.append(instance['DBInstanceIdentifier'])
-            rds_instance_source = session.query(RdsInstanceSource).filter(
-                RdsInstanceSource.instance_id == instance['DBInstanceIdentifier'],
-                RdsInstanceSource.region == region,
-                # RdsInstanceSource.instance_class == instance['DBInstanceClass'],
-                # RdsInstanceSource.engine == instance['Engine'],
-                # RdsInstanceSource.instance_status == instance['DBInstanceStatus'],
-                # RdsInstanceSource.address == instance['Endpoint']['Address'] if 'Endpoint' in instance else '',
-                # RdsInstanceSource.port == instance['Endpoint']['Port'] if 'Endpoint' in instance else '',
-                # RdsInstanceSource.master_username == instance['MasterUsername'],
-                RdsInstanceSource.aws_account == aws_account_id,
-            ).scalar()
-            if rds_instance_source is None:
-                rds_instance_source = RdsInstanceSource(instance_id=str(instance['DBInstanceIdentifier']),
-                                                        region=region,
-                                                        instance_class=instance['DBInstanceClass'],
-                                                        engine=instance['Engine'],
-                                                        instance_status=instance['DBInstanceStatus'],
-                                                        address=instance['Endpoint']['Address'],
-                                                        port=instance['Endpoint']['Port'],
-                                                        master_username=instance['MasterUsername'],
-                                                        # created_time=instance['Engine'],
-                                                        aws_account=aws_account_id
+        glue_database_list.append(client.get_databases()['DatabaseList'])
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The API method `get_databases` returns paginated results instead of all results. Consider using the pagination API or checking one of the following keys in the response to verify that all results were returned: `IsTruncated`, `NextToken`.

[Learn more](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/paginators.html)